test/zipkin/util.go

```diff
@@ -0,0 +1,139 @@
+// b) enable AlwaysSample config for tracing.
+func SetupZipkinTracing(kubeClientset *kubernetes.Clientset) error {
+
+	logger := logging.GetContextLogger("SpoofUtil")
+
+	zipkinPortAvailable, err := CheckZipkinPortAvailability()
+	if !zipkinPortAvailable {
+		return fmt.Errorf("Zipkin port not available on the machine: %v. Zipkin Setup failed", err)
+	}
+
+	zipkinPods, err := kubeClientset.CoreV1().Pods(ZipkinNamespace).List(metav1.ListOptions{LabelSelector: "app=zipkin"})
+
+	if err != nil {
+		return fmt.Errorf("Error retrieving Zipkin pod details : %v. Zipkin Setup failed", err)
+	}
+
+	if len(zipkinPods.Items) == 0 {
+		return errors.New("No Zipkin Pod found on the cluster. Ensure monitoring is switched on for your Knative Setup. Zipkin Setup failed")
+	}
+
+	portForwardCmd := exec.Command("kubectl", "port-forward", "--namespace=" + ZipkinNamespace, zipkinPods.Items[0].Name, fmt.Sprintf("%d:%d", ZipkinPort, ZipkinPort))
+	err = portForwardCmd.Start()
+
+	if err != nil {
+		return fmt.Errorf("Error starting kubectl port-forward command : %v", err)
+
+	}
+
+	os.Setenv(ZipkinPortForwardPid, strconv.Itoa(portForwardCmd.Process.Pid))
+
+	// Applying AlwaysSample config to ensure we propagate zipkin header for every request made by this client.
+	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+
+	logger.Infof("Successfully setup SpoofingClient for Zipkin Tracing")
+
+	return nil
+}
+
+// CleanupZipkinTracingSetup cleans up the Zipkin tracing setup on the machine. This involves killing the process performing port-forward.
+// Method expects the port-forwarding process to be stored in an environment variable.
+func CleanupZipkinTracingSetup() error {
+
+	logger := logging.GetContextLogger("SpoofUtil")
+	pid := os.Getenv(ZipkinPortForwardPid)
+
+	if len(pid) == 0 {
+		return fmt.Errorf("Zipking port-forward Process ID not found. CleanupZipkingTracingSetup() expects %s environment variable to be pointing to Process ID perform port-forwarding", ZipkinPortForwardPid)
+	}
+
+	logger.Infof("Found Zipkin port-forward Process ID: %s", pid)
+
+	pidInt, err := strconv.Atoi(pid)
+
+	if err != nil {
+		return fmt.Errorf("Error converting Zipkin port-forward Process ID into an Int : %v", err)
+	}
+
+	ps := os.Process{Pid: pidInt}
+	err = ps.Kill()
+
+	if err != nil {
+		return fmt.Errorf("Encoutered error killing port-forward process in CleanupZipkingTracingSetup(..) : %v", err)
+	}
+
+	logger.Infof("Successfully killed Zipkin port-forward process")
+
+	return nil
+}
+
+// CheckZipkinPortAvailability checks to see if Zipkin Port is available on the machine.
+// If due to error it is not possible to determine if the port is available the method returns "false"
+func CheckZipkinPortAvailability() (bool, error) {
+
+	host := ":" + strconv.Itoa(ZipkinPort)
+	server, err := net.Listen("tcp", host)
```

Comment:
can be combined.

if server, err := net.Listen(); err != nil {
return false, err
} else {
server.Close()
return true, nil
}

Reply:
As I mentioned before, this is unnecessary complexity. Just return the error or `nil`. You don't even need `host` here, because it's used just once and is a simple concatenation (I would even use `fmt.Sprintf` for clarity).

Reply:
Fixed.